### PR TITLE
doc: add note about Python version

### PR DESCRIPTION
**Pull request by eddieantonio**
Document the minimum version of Python required. Thanks @tobyatgithub!

### CODE REVIEW

**Review by joanise — CHANGES_REQUESTED** [review]
@eddieantonio Thanks for doing this, it should definitely be documented! I'm sure the answer to my next question is yes: is there a line of code we could add to `cli.py` and `app.py` to say "requires 3.7" that would exit immediately with a clear error message in older Pythons? (My syntax is approximate Perl, I'm sure the same exists in Python.)

**Comment by eddieantonio** [comment]
> @eddieantonio Thanks for doing this, it should definitely be documented! I'm sure the answer to my next question is yes: is there a line of code we could add to `cli.py` and `app.py` to say "requires 3.7" that would exit immediately with a clear error message in older Pythons? (My syntax is approximate Perl, I'm sure the same exists in Python.)

It is indeed possible, though doesn't have neat, straightforward syntax like Perl (yes, I just complimented Perl's syntax!)

File: readalongs/python_version.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+def ensure_using_supported_python_version():
+    if sys.version_info < MINIMUM_PYTHON_VERSION_REQUIRED:
+        sys.exit(
+            "Python "
+            + _version_to_str(MINIMUM_PYTHON_VERSION_REQUIRED)
+            + " required (you are currently using Python "
+            + _version_to_str(sys.version_info)
+            + "). Please use a newer version of Python."
+        )
```
Comment (eddieantonio) [inline readalongs/python_version.py:19]: Unfortunately, there is no `require v3.7`; however, this has a similar effect to the Perl idiom; exiting with an error status and no traceback when the version is too old.

Comment (joanise) [reply readalongs/python_version.py:19]: Wow, it's just silly there isn't a briefer way to do this! Thanks for providing the solution!

**Review by joanise — APPROVED** [review]

**Pull request merged**